src/algorithms/GenericFunctions.jl

```diff
@@ -0,0 +1,31 @@
+###############################################################################
+#
+#   GenericFunctions.jl : Functions for Generic types
+#
+###############################################################################
+
+# TODO: Move from CRT from Hecke to AbstractAlgebra?
+# Currently no implementation, only example on how the arbitrary inputs `crt`
+# should look like.
+@doc Markdown.doc"""
+    crt(r::Array{T, 1}, m::Array{T, 1}) where T
+    crt(r::T, m::T...) where T
+
+Return $x$ in the Euclidean domain $T$ such that $x \equiv r_i \mod m_i$
+for all $i$.
+"""
+function crt end
+
+@doc Markdown.doc"""
+    factor(a::T, b::R)
+
+Return factorization of element $a$. 
+"""
+function factor end
+
+@doc Markdown.doc"""
+    factor_squarefree(a::T)
+
+Return square free factorization of element $a$.
```

Comment:
What kind of element is `a` allowed to be? An element of a commutative factorization domain? Or more general? 

And why does `factor` take a ring argument but `factor_squarefree` does not? I mean, either the type of `a` determines the type, then both don't need it; or it does not determine the type / the ability to embed `a` into a different ring without coercing it is strongly desired (but why?) but then this should perhaps be mentioned? Or thirdly, I am misunderstanding something (wouldn't surprise me a bit ;-) ).

Reply:
Hmm... Since they were functions in `generic_functions.jl` at first, they were kept initially when moving everything to `fundamental_interface.jl`. But they don't fit there, they are more type specific functions, so I moved them back.

I have no knowledge of these types of things, so maybe someone can come with a suggested docstring? I basically looked at `Hecke` to write some type of general docstring.

Reply:
OK, fair enough, thanks for the clarification! Perhaps @thofma or @fieker can clarify then.

Reply:
Maybe they shoud not have a docstring. They are defined here in AA so they can be extended.